### PR TITLE
feat: Updated distillation workflow to store and load (top-k) logits

### PR DESCRIPTION
**Pull request by HannanJaved**
distill.py updated to load stored logits; store_teacher_logits.py to store (top-k) logits

#### Reference Issues/PRs
#262 

#### What does this implement/fix? Explain your changes.
Added store_teacher_logits.py to store (top-k) logits from a teacher model
Updated distill.py to load the stored teacher logits for distillation

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

File: whittle/distill.py [hunk 1]
```diff
@@ -270,19 +297,65 @@ def main(
...
+                        checkpoint = os.path.join(teacher_checkpoint_dir, "lit_model.pth")
+                        temp_teacher = fabric.setup(temp_teacher)
+                        load_checkpoint(fabric, temp_teacher, checkpoint, strict=False)
+                else:
+                    temp_teacher = teacher
+
+                temp_teacher.select_sub_network(subnetwork)
+                student = extract_current_sub_network(temp_teacher)
+
+                if random_init_student:
```
[the remark this answers is not in the document]
Comment (HannanJaved) [reply whittle/distill.py:336]: Yes, makes sense. I will change this part

File: whittle/distill.py [hunk 2]
```diff
@@ -390,6 +478,117 @@ def main(
...
+        self._current_input_ids = None
+        self._load_batch(0)
+
+    def _load_batch(self, batch_idx: int):
+        """Load a specific batch of logits to CPU first."""
+        if batch_idx >= self.num_batches:
+            return
+
+        # Load to CPU first to avoid OOM
+        logits_data = torch.load(self.logits_files[batch_idx], map_location="cpu")
```
[the remark this answers is not in the document]
Comment (HannanJaved) [reply whittle/distill.py:519]: The size of the stored logits batches could be larger than the batch size used for training so wanted load all of them to cpu and then move only the relevant slices of that batch to GPU when needed.

**Comment by HannanJaved** [comment]
Created issue #333 to work on the code quality changes

**Review by aaronkl — APPROVED** [review]
